### PR TITLE
LibWeb+LibJS: Implement the js-api portion of Wasm multi-value

### PR DESCRIPTION
**Pull request by alimpfard**
@ADKaster made me do this yesterday in a silly way, so here's the proper implementation :^)

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/ErrorTypes.h [hunk 1]
```diff
@@ -327,7 +327,8 @@
                                                       "must be called with the RegExp constructor for the this value")                  \
     M(GetLegacyRegExpStaticPropertyThisValueMismatch, "Legacy RegExp static property getter "                                           \
                                                       "must be called with the RegExp constructor for the this value")                  \
-    M(GetLegacyRegExpStaticPropertyValueEmpty, "Legacy RegExp static property getter value is empty")
+    M(GetLegacyRegExpStaticPropertyValueEmpty, "Legacy RegExp static property getter value is empty")                                   \
+    M(InvalidWasmMultiValueResultCount, "Invalid number of return values for multi-value wasm return of {} objects")
```
Comment (linusg) [inline Userland/Libraries/LibJS/Runtime/ErrorTypes.h:331]: Please don't put LibWeb error messages in LibJS, inline the string for now or start a similar enumeration over there

File: Userland/Libraries/LibWeb/WebAssembly/WebAssemblyObject.cpp [hunk 2]
```diff
@@ -464,5 +480,4 @@ WebAssemblyMemoryObject::WebAssemblyMemoryObject(JS::Realm& realm, Wasm::MemoryA
     , m_address(address)
 {
 }
-
```
Comment (linusg) [inline Userland/Libraries/LibWeb/WebAssembly/WebAssemblyObject.cpp:475]: Ctrl+Z

File: Userland/Libraries/LibWasm/CMakeLists.txt [hunk 3]
```diff
@@ -8,7 +8,7 @@ set(SOURCES
 )
 
 serenity_lib(LibWasm wasm)
-target_link_libraries(LibWasm PRIVATE LibCore)
+target_link_libraries(LibWasm PRIVATE LibCore LibJS)
```
Comment (linusg) [inline Userland/Libraries/LibWasm/CMakeLists.txt:11]: Seems kinda wrong to connect LibWasm to LibJS outside of the LibWeb glue code, but I assume you know what you're doing so...

File: Userland/Libraries/LibJS/Runtime/ErrorTypes.h (hunk 1 above)
Comment (alimpfard) [reply Userland/Libraries/LibJS/Runtime/ErrorTypes.h:331]: Alright.

File: Userland/Libraries/LibWasm/CMakeLists.txt (hunk 3 above)
Comment (alimpfard) [reply Userland/Libraries/LibWasm/CMakeLists.txt:11]: Not much else I can do, I could skip the libjs link if I skip using Value's methods, but that's only really gonna work until someone decides to move something in `JS::Completion` out of line; so might as well just link it and call it good.

Comment (ADKaster) [reply Userland/Libraries/LibWasm/CMakeLists.txt:11]: That is, unless you want to do weird stuff like move the JS bindings into a separate library, or into LibJS

File: Tests/LibWasm/test-wasm.cpp [hunk 4]
```diff
@@ -252,15 +252,24 @@ JS_DEFINE_NATIVE_FUNCTION(WebAssemblyModule::wasm_invoke)
...
+    };
+
+    if (result.values().size() == 1)
+        return to_js_value(result.values().first());
+
+    Vector<JS::Value> values;
+    values.ensure_capacity(result.values().size());
+    for (auto& value : result.values())
+        values.append(to_js_value(value));
+    return JS::Array::create_from(*vm.current_realm(), move(values));
```
Comment (linusg) [inline Tests/LibWasm/test-wasm.cpp:274]: Would be nicer as:

```cpp
auto& realm = *vm.current_realm();
...
return Array::create_from<Wasm::Value>(realm, result.values(), [](auto& value) {
    return to_js_value(value);
});
```

**Review by linusg — COMMENTED** [review]
one more, rest LGTM

File: Tests/LibWasm/test-wasm.cpp (hunk 4 above)
Comment (alimpfard) [reply Tests/LibWasm/test-wasm.cpp:274]: Oooh, didn't know about that overload. Sure.

**Pull request merged**